Products/GenericSetup/tool.py

```diff
@@ -921,17 +921,30 @@ def getProfileInfo(self, profile_id):
 
     @security.protected(ManagePortal)
     def getDependenciesForProfile(self, profile_id):
-        if profile_id is None:
-            return ()
-        if profile_id.startswith("snapshot-"):
-            return ()
 
-        if not self.profileExists(profile_id):
-            raise KeyError(profile_id)
-        try:
-            return self.getProfileInfo(profile_id).get('dependencies', ())
-        except KeyError:
-            return ()
+        dependencies = None
+        profile_info = None
+
+        if profile_id is None or profile_id.startswith("snapshot-"):
+            dependencies = ()
+
+        profile_info = self.getProfileInfo(profile_id)
```

Comment:
I'm assuming that you want the KeyError raised by this line to propagate:  is that correct?